### PR TITLE
Page to import entities from a JSON file

### PR DESCRIPTION
**Pull request by helgridly**

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/main.cljs [hunk 1]
```diff
@@ -57,6 +58,9 @@
   {:render
    (fn [{:keys [props]}]
      [:div {:style {:textAlign "right"}}
+      [TopNavBarLink {:name "Import Data"
```
Comment (helgridly) [inline src/cljs/org/broadinstitute/firecloud_ui/main.cljs]: So if you're adding something to the top nav, you have to do so in three places: here, line 77, and line 96. Can we make a list of "things to put in the top nav" and reference it from these three places instead? 'cos I missed one first time round.

Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/main.cljs]: That's a good idea. Something like:
```clojure
(def top-nav-bar-items
[{:key :workspaces
  :name "Workspaces"
  :component workspaces/Page}
  ...
]
```
should work. Feel free to go for it or just add a TODO for us to tackle.

Comment (helgridly) [inline src/cljs/org/broadinstitute/firecloud_ui/main.cljs]: I'll do it in a separate PR.

File: src/cljs/org/broadinstitute/firecloud_ui/utils.cljs [hunk 2]
```diff
@@ -93,3 +93,10 @@
     (fn [x1 x2] (if (and (map? x1) (map? x2)) (deep-merge x1 x2) x2))
     maps))
 
+
+(defn generate-form-data [params]
+  "Create a blob of multipart/form-data from the provided map."
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/utils.cljs]: Annoyingly, the docs for `defn` have the doc string directly after the name and before the parameters. We should probably follow this even though it sucks.

File: src/cljs/org/broadinstitute/firecloud_ui/page/import_data.cljs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  (utils/ajax-orch
+    (str "/workspaces/" workspaceNamespace "/" workspaceName "/importEntitiesJSON")
+    {:method :post
+     :encType "multipart/form-data"
+     :on-done :default
+     :data (utils/generate-form-data {:entities entityFile})}))
+
+
+(react/defc Page
+            {:render
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/import_data.cljs]: Nit: This should be a normal (2-space) indent. I'll try to figure out how to make IntelliJ behave.

File: src/cljs/org/broadinstitute/firecloud_ui/utils.cljs (hunk 2 above)
Comment (helgridly) [inline src/cljs/org/broadinstitute/firecloud_ui/utils.cljs]: Yeah, I'd read this but I copied off something in utils for consistency.

File: src/cljs/org/broadinstitute/firecloud_ui/page/import_data.cljs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+(ns org.broadinstitute.firecloud-ui.page.import-data
+  (:require
+    [clojure.string]
+    [dmohs.react :as react]
+    [org.broadinstitute.firecloud-ui.common :as common]
+    [org.broadinstitute.firecloud-ui.common.components :as comps]
+    [org.broadinstitute.firecloud-ui.common.style :as style]
+    [org.broadinstitute.firecloud-ui.common.table :as table]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/import_data.cljs]: We don't have optimize imports. Sorry. No need to clean these up, but do so if you feel like it.

**Comment by dmohs** [comment]
Good to merge as-is or tweak and merge.

File: src/cljs/org/broadinstitute/firecloud_ui/page/import_data.cljs (hunk 3 above)
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/import_data.cljs]: Preferences -> Editor -> Code Style -> Clojure -> General tab -> "Default to Only Indent" should be checked.

**Pull request merged**